### ISSUE TITLE
Return type of return_order_book should not be a Map

### ISSUE THREAD

**Issue by doomsplayer** [hugues31/coinnect#65]
Actually, not all return value from Poloniex is a JSON object, for now, `return_order_book` is an exception. I make a change to these methods to return `Result<T>`, which delegates the choice to the user that he can decide what type the result value is. [Commit](https://github.com/doomsplayer/coinnect/commit/50dcd87f85e105ba7f84b020eb1afdaa2691999e)

**Comment by hugues31** [comment]
Using the Deserialize from Serde seems a great idea. However does it break the generic API ? I didn't run tests to check it.

**Comment by doomsplayer** [comment]
I don't think so. Since the generic API assumes the return type as a `Map`, so the compiler will infer `T` as a `Map`. Then everything falls back to the original `Map` API.

**Comment by doomsplayer** [comment]
Actually using generic is just a workaround. Finally, I think we need to type-ify all the return types. Something like `https://transform.now.sh/json-to-rust-serde` is pretty useful.

**Comment by hugues31** [comment]
Yes but I think that a strong typed return should be the purpose of the generic API, the raw API, which is lower level, should stay close to the exchange response. Eventually, the Map container may be changed.
